libs/live-dmk-mobile/src/transport/DeviceManagementKitHIDTransport.ts

```diff
@@ -0,0 +1,217 @@
+    } catch (err) {
+      tracer.trace("[open] open error", { err });
+      throw err;
+    }
+  }
+
+  static listen(
+    observer: TransportObserver<DescriptorEvent<string>, HwTransportError>,
+    _context?: TraceContext,
+    dmk: DeviceManagementKit = getDeviceManagementKit(),
+  ) {
+    let availableSubscription: Subscription | undefined = undefined;
+
+    availableSubscription = dmk
+      .listenToAvailableDevices({
+        transport: rnHidTransportIdentifier,
+      })
+      .pipe(startWith<DiscoveredDevice[]>([]), pairwise())
+      .subscribe({
+        next: ([prevDevices, currDevices]) => {
+          const removedDevice = prevDevices.find(
+            prevDevice => !currDevices.some(device => device.id === prevDevice.id),
+          );
+          if (removedDevice) {
+            observer.next({
+              type: "remove",
+              descriptor: removedDevice.id,
+              device: removedDevice,
+            });
+          }
+          for (const device of currDevices) {
+            const deviceModelId = dmkToLedgerDeviceIdMap[device.deviceModel.model];
+            const deviceModel = getDeviceModel(deviceModelId);
+            observer.next({
+              type: "add",
+              descriptor: device.id,
+              device: device,
+              deviceModel,
+            });
+          }
+        },
+        complete: observer.complete,
+        error: err => {
+          tracer.trace("[listen] error", err);
+          observer.error(err);
+          availableSubscription?.unsubscribe();
+        },
+      });
+
+    const unsubscribe = () => {
+      availableSubscription?.unsubscribe();
+    };
+
+    return {
+      unsubscribe,
+    };
+  }
+
+  async exchange(apdu: Buffer, { abortTimeoutMs }: { abortTimeoutMs?: number } = {}) {
+    const activeSessionId = activeDeviceSessionSubject.value?.transport.sessionId;
+    if (!activeSessionId) {
+      throw new DisconnectedDevice();
+    }
+
+    return await this.dmk
+      .sendApdu({
+        sessionId: activeSessionId,
+        apdu: new Uint8Array(apdu),
+        abortTimeout: abortTimeoutMs,
+      })
+      .then((apduResponse: { data: Uint8Array; statusCode: Uint8Array }): Buffer => {
+        const response = Buffer.from([...apduResponse.data, ...apduResponse.statusCode]);
+        return response;
+      })
+      .catch(error => {
+        tracer.trace("[DMKTransportHID] [exchange] error", { error });
+        if (
+          error instanceof SendApduEmptyResponseError ||
+          error instanceof DeviceDisconnectedWhileSendingError ||
+          error instanceof DeviceDisconnectedBeforeSendingApdu
+        )
+          throw new DisconnectedDevice();
+        throw error;
+      });
+  }
+
+  close() {
+    return Promise.resolve();
+  }
+
+  listenToDisconnect() {
+    let subscription: Subscription | undefined = undefined;
+
+    subscription = this.dmk.getDeviceSessionState({ sessionId: this.sessionId }).subscribe({
```

Comment:
You might consider moving the if (state.deviceStatus === DeviceStatus.NOT_CONNECTED) check into the RxJS pipeline using the filter operator.

`  .pipe(filter(state => state.deviceStatus === DeviceStatus.NOT_CONNECTED))`